src/bin/metadata.rs

```diff
@@ -0,0 +1,54 @@
+extern crate cargo;
+extern crate docopt;
+extern crate rustc_serialize;
+extern crate toml;
+
+
+use cargo::ops::{output_metadata, OutputTo, OutputFormat, OutputMetadataOptions};
+use cargo::util::{CliResult, CliError, Config};
+use cargo::util::important_paths::find_root_manifest_for_wd;
+
+#[derive(RustcDecodable)]
+struct Options {
+    flag_features: Vec<String>,
+    flag_manifest_path: Option<String>,
+    flag_no_default_features: bool,
+    flag_output_format: OutputFormat,
+    flag_output_path: OutputTo,
```

Comment:
Perhaps this could just be `Option<String>`? That way a custom `Decodable` implementation isn't needed